cudapoa/src/cudapoa_nw_banded.cu

```diff
@@ -197,16 +200,16 @@ __device__
     const ScoreT min_score_value              = 2 * abs(min(min(gap_score, mismatch_score), -match_score) - 1) + score_type_min_limit;
 
     int16_t lane_idx = threadIdx.x % WARP_SIZE;
+    int64_t score_index;
 
     //Calculate gradient for the scores matrix
     float gradient = float(read_length + 1) / float(graph_count + 1);
 
-    SizeT band_width = WARP_SIZE * CELLS_PER_THREAD;
-
-    SizeT max_column = read_length + 1;
+    SizeT max_column                    = read_length + 1;
+    SizeT max_matrix_sequence_dimension = band_width + CUDAPOA_BANDED_MATRIX_RIGHT_PADDING;
```

Comment:
I don't recall now what the `CUDAPOA_BANDED_MATRIX_RIGHT_PADDING` was for. It looks like it has to be added to every indexing parameter. can you remind me what it was for? It could become a source of bugs if it's not added carefully to each index variable, so I'm wondering if there's a way to get rid of it...

Reply:
from what I understand, bands per row are centered around diagonal. Depending on the shape of the NW matrix (how tall or fat it is), the location of the band in each row can shift up to a couple of cells to the right relative to the band location of the previous row.
    __________________________                            __________________________----
        ___________________________           =>       __________________________----
            ____________________________                 __________________________----          

 In the banded score matrix, for processing elements toward the end of a row, we need to access to its upper adjacent cells, which due to shifting, if not padded with extra length, it will be invalid memory access. So either we can use if statement for those end cells which results in branch divergence, or pad the score matrix with some extra length.

Reply:
Thanks for the explanation, @r-mafi !